pr-review engine: `COPILOT_API_MODEL` unbound var silently disables the cross-engine rubber-duck tier (engine.sh:431)

## Summary
On every `claude`-primary PR review (the default engine), the **cross-engine rubber-duck (tier-2) is silently disabled** by an `unbound variable` error. The cascade falls back to deep-review-only, so the o4-mini rubber-duck cross-check never runs.

## Evidence (live v1.8.0 / ring-0 run)
Observed in `pr-review-trigger` run [`27904795931`](https://github.com/petry-projects/.github-private/actions/runs/27904795931) (job `82571409998`), 2026-06-21, which ran the agent pinned at `pr-review/next` (`ded84ce` = v1.8.0). The run still concluded **success**, masking the defect:

```
scripts/engine.sh: line 431: COPILOT_API_MODEL: unbound variable
[tier2] rubber duck did not produce valid JSON — continuing with deep review only
```

## Root cause
- `set_engine_config`'s `claude)` branch sets the cross-engine duck to copilot: `DUCK_ENGINE="copilot"` / `DUCK_MODEL="o4-mini"` (`scripts/engine.sh:87-89`).
- But `COPILOT_API_MODEL`'s default is assigned **only inside the `copilot)` branch**: `COPILOT_API_MODEL="${COPILOT_API_MODEL:-openai/o4-mini}"` (`scripts/engine.sh:147`).
- The rubber-duck path invokes `_copilot_invoke`, which references `$COPILOT_API_MODEL` at `scripts/engine.sh:431` (and `:436`).
- With `REVIEW_ENGINE=claude` (default), the `copilot)` branch never runs, so `COPILOT_API_MODEL` is unset. Under `set -euo pipefail` (`set -u`), the reference aborts the duck subprocess → invalid JSON → tier-2 skipped.

Note: `DUCK_MODEL="o4-mini"` is set, but `_copilot_invoke` keys off `COPILOT_API_MODEL` (the GitHub Models API id, e.g. `openai/o4-mini`), not `DUCK_MODEL` — so setting the duck model alone doesn't cover this path.

## Impact
- Cross-engine rubber-duck (independent o4-mini second opinion) is **off on all claude-primary reviews** — reduced review coverage, no error surfaced to the run conclusion (it stays green).
- Latent for any non-copilot primary that uses the copilot duck.

## Suggested fix
Give `COPILOT_API_MODEL` an unconditional default so it's defined whenever the copilot engine is invoked (primary **or** rubber-duck), e.g. hoist the default out of the `copilot)` branch (set it alongside the other unconditional exports near `scripts/engine.sh:167-170`), or default at point-of-use in `_copilot_invoke` (`${COPILOT_API_MODEL:-openai/o4-mini}`). Add a regression test asserting that a `claude`-primary config leaves `COPILOT_API_MODEL` bound (and that the duck path doesn't trip `set -u`).

_Found while verifying v1.8.0 ring-0 run health._

## Dev-Lead: Implementation Complete

### Plan Execution
- [x] `scripts/engine.sh` — hoisted the `COPILOT_API_MODEL="${COPILOT_API_MODEL:-openai/o4-mini}"` default out of the `copilot)` branch so it is set **unconditionally** after the `case`, before the export block. The existing `export ... COPILOT_API_MODEL` line now always exports a bound value.
- [x] `tests/dev-lead/unit/test_engine_duck_model.bats` — new regression suite (5 tests).

### Root cause / fix
The `claude)` branch wires the cross-engine duck to copilot (`DUCK_ENGINE=copilot`), but `COPILOT_API_MODEL`'s default was assigned **only** inside the `copilot)` branch. Under a `claude`-primary review (the default) that branch never runs, leaving `COPILOT_API_MODEL` unset; `copilot_chat`'s bare `$COPILOT_API_MODEL` reference (engine.sh:431/436) then aborted under `set -u`, silently skipping tier-2. The default is now unconditional, so the duck path stays bound for any non-copilot primary.

### Test Results
New regression suite (red before fix on tests 1/2/5, green after):
```
1..5
ok 1 duck-model: claude-primary config leaves COPILOT_API_MODEL bound to default
ok 2 duck-model: sourcing as claude under set -u does not trip on COPILOT_API_MODEL (engine.sh:431 regression)
ok 3 duck-model: explicit COPILOT_API_MODEL override is honored under claude
ok 4 duck-model: copilot-primary still binds COPILOT_API_MODEL to default (no regression)
ok 5 duck-model: gemini-primary also leaves COPILOT_API_MODEL bound (latent non-copilot duck path)
```
Full dev-lead unit suite: **577 passed, 0 failed** (`bats tests/dev-lead/unit/`).

### Files Changed
- `scripts/engine.sh`: moved the `COPILOT_API_MODEL` default + explanatory comment out of the `copilot)` branch to an unconditional assignment in `set_engine_config` (before the exports). No model values changed; copilot-primary behavior is byte-identical.
- `tests/dev-lead/unit/test_engine_duck_model.bats`: new — asserts claude/gemini/copilot primaries all leave `COPILOT_API_MODEL` bound to `openai/o4-mini`, that an explicit override is honored, and that sourcing as `claude` under `set -u` + a bare `$COPILOT_API_MODEL` reference does not abort.

### Verification
- `bash .dev-lead/scripts/dev-lead-lint.sh` → all checks passed (shellcheck on 72 files + agent frontmatter + CODEOWNERS).
- `shellcheck scripts/engine.sh` → exit 0 (only the two pre-existing SC2015/SC1091 *info* notes at line 276, unrelated to this change).

### Notes
- `scripts/review-batch.sh` already guards its reads with `${COPILOT_API_MODEL:-...}`, so it was unaffected.
- Pre-existing unrelated failure: `tests/test_batch_fallback.bats` test "Gemini trust error (55) falls back to Copilot" fails on the **clean** tree as well (verified via `git checkout scripts/engine.sh`); it concerns gemini fallback, not `COPILOT_API_MODEL`, and is out of scope for #881.

--- a/scripts/engine.sh
+++ b/scripts/engine.sh
@@ -163,7 +163,7 @@ set_engine_config() {
   # openai/o4-mini is the April-2025 o4-generation reasoning model; it is not a
   # typo for o1-mini or gpt-4o-mini.
   #
-  # Defaulted UNCONDITIONALLY (not just in the copilot) branch) because the
+  # Defaulted UNCONDITIONALLY (not just in the copilot branch) because the
   # cross-engine rubber-duck uses copilot even when it is not primary — e.g. the
   # claude) branch sets DUCK_ENGINE=copilot — and copilot_chat references a bare
   # $COPILOT_API_MODEL under `set -u`. Leaving it unset on a non-copilot primary

--- a/scripts/engine.sh
+++ b/scripts/engine.sh
@@ -139,13 +139,6 @@ set_engine_config() {
       ENGINE_AUDIT_MODEL="o4-mini"
       ENGINE_ACTION_MODEL="o4-mini"
       ENGINE_SINGLE_MODEL="o4-mini"
-      # GitHub Models API model identifier — must match a model available at
-      # https://models.github.ai (see GitHub Models marketplace).
-      # Override via COPILOT_API_MODEL env var if the default is unavailable.
-      # openai/o4-mini is the April-2025 o4-generation reasoning model; it is
-      # not a typo for o1-mini or gpt-4o-mini.
-      COPILOT_API_MODEL="${COPILOT_API_MODEL:-openai/o4-mini}"
-      export COPILOT_API_MODEL
       ENGINE_LABEL="triage: o4-mini → deep: o4-mini + duck: gemini-3.5-flash → audit: o4-mini (GitHub Models API)"
       ENGINE_SINGLE_LABEL="single-reviewer mode: o4-mini (GitHub Models API)"
       # Cross-engine rubber duck: use Gemini when Copilot is primary
@@ -163,6 +156,19 @@ set_engine_config() {
       exit 1
       ;;
   esac
+
+  # GitHub Models API model identifier for the copilot engine — must match a
+  # model available at https://models.github.ai (see GitHub Models marketplace).
+  # Override via COPILOT_API_MODEL env var if the default is unavailable.
+  # openai/o4-mini is the April-2025 o4-generation reasoning model; it is not a
+  # typo for o1-mini or gpt-4o-mini.
+  #
+  # Defaulted UNCONDITIONALLY (not just in the copilot) branch) because the
+  # cross-engine rubber-duck uses copilot even when it is not primary — e.g. the
+  # claude) branch sets DUCK_ENGINE=copilot — and copilot_chat references a bare
+  # $COPILOT_API_MODEL under `set -u`. Leaving it unset on a non-copilot primary
+  # silently aborts the duck subprocess and skips tier-2 (#881).
+  COPILOT_API_MODEL="${COPILOT_API_MODEL:-openai/o4-mini}"
 
   export ENGINE_TRIAGE_MODEL ENGINE_DEEP_MODEL ENGINE_AUDIT_MODEL
   export ENGINE_ACTION_MODEL ENGINE_SINGLE_MODEL

--- a/tests/dev-lead/unit/test_engine_duck_model.bats
+++ b/tests/dev-lead/unit/test_engine_duck_model.bats
@@ -1,0 +1,58 @@
+#!/usr/bin/env bats
+# Regression tests for engine.sh — COPILOT_API_MODEL must stay bound for the
+# cross-engine rubber-duck path even when copilot is NOT the primary engine.
+#
+# Bug (#881): the claude) branch wires the duck to copilot (DUCK_ENGINE=copilot)
+# but COPILOT_API_MODEL's default was assigned ONLY inside the copilot) branch.
+# Under a claude-primary review (the default), COPILOT_API_MODEL stayed unset, so
+# copilot_chat's bare `$COPILOT_API_MODEL` reference (engine.sh:431/436) aborted
+# under `set -u` → invalid duck JSON → tier-2 silently skipped.
+
+SCRIPT_DIR="$(cd "$(dirname "$BATS_TEST_FILENAME")/../../.." && pwd)"
+ENGINE_SCRIPT="$SCRIPT_DIR/scripts/engine.sh"
+
+setup() {
+  # Clear any inherited value so engine.sh defaults are evaluated fresh.
+  unset COPILOT_API_MODEL
+}
+
+teardown() {
+  unset COPILOT_API_MODEL
+}
+
+_source_engine() {
+  local engine="${1:-claude}"
+  export REVIEW_ENGINE="$engine"
+  source "$ENGINE_SCRIPT"
+}
+
+@test "duck-model: claude-primary config leaves COPILOT_API_MODEL bound to default" {
+  _source_engine "claude"
+  # ${VAR-x} (no colon) distinguishes unset from empty: unset → marker.
+  [ "${COPILOT_API_MODEL-__UNSET__}" = "openai/o4-mini" ]
+}
+
+@test "duck-model: sourcing as claude under set -u does not trip on COPILOT_API_MODEL (engine.sh:431 regression)" {
+  # Reproduce the production failure: a bare reference to $COPILOT_API_MODEL
+  # under `set -u` (which engine.sh enables) must not abort.
+  run bash -c "set -euo pipefail; unset COPILOT_API_MODEL; REVIEW_ENGINE=claude; source '$ENGINE_SCRIPT'; echo \"model=\$COPILOT_API_MODEL\""
+  [ "$status" -eq 0 ]
+  [[ "$output" == *"model=openai/o4-mini"* ]]
+  [[ "$output" != *"unbound variable"* ]]
+}
+
+@test "duck-model: explicit COPILOT_API_MODEL override is honored under claude" {
+  export COPILOT_API_MODEL="openai/gpt-5-mini"
+  _source_engine "claude"
+  [ "$COPILOT_API_MODEL" = "openai/gpt-5-mini" ]
+}
+
+@test "duck-model: copilot-primary still binds COPILOT_API_MODEL to default (no regression)" {
+  _source_engine "copilot"
+  [ "${COPILOT_API_MODEL-__UNSET__}" = "openai/o4-mini" ]
+}
+
+@test "duck-model: gemini-primary also leaves COPILOT_API_MODEL bound (latent non-copilot duck path)" {
+  _source_engine "gemini"
+  [ "${COPILOT_API_MODEL-__UNSET__}" = "openai/o4-mini" ]
+}

